src/pybind/mgr/ssh/remotes.py

```diff
@@ -0,0 +1,87 @@
+# ceph-deploy ftw
+import os
+import errno
+import tempfile
+import shutil
+
+def safe_makedirs(path, uid=-1, gid=-1):
+    """ create path recursively if it doesn't exist """
+    try:
+        os.makedirs(path)
+    except OSError as e:
+        if e.errno == errno.EEXIST:
+            pass
+        else:
+            raise
+    else:
+        os.chown(path, uid, gid)
+
+def write_conf(path, conf):
+    if not os.path.exists(path):
+        dirpath = os.path.dirname(path)
+        if os.path.exists(dirpath):
+            with open(path, "w") as f:
+                f.write(conf)
+            os.chmod(path, 0o644)
+        else:
+            raise RuntimeError(
+                "{0} does not exist".format(dirname))
+
+def write_keyring(path, key, overwrite=False, uid=-1, gid=-1):
+    dirname = os.path.dirname(path)
+    if not os.path.exists(dirname):
+        makedir(dirname, uid, gid)
+    if not overwrite and os.path.exists(path):
+        return
+    with open(path, "wb") as f:
+        f.write(key.encode('utf-8'))
+
+def create_mon_path(path, uid=-1, gid=-1):
+    """create the mon path if it does not exist"""
+    if not os.path.exists(path):
+        os.makedirs(path)
+        os.chown(path, uid, gid);
+
+def path_exists(path):
```

Comment:
this is used nowhere. and probably we can drop this very thin wrapper?

Reply:
yeh, that was dead code. the thin wrappers seem to be necessary to work with remoto. these methods are actually invoked remotely over ssh, and in this case, its just proxying some `os.xyz` calls.